project/src/components/offers-list/offers-list.tsx

```diff
@@ -1,22 +1,28 @@
+import { Fragment } from 'react';
+import {Offer} from '../../types/offer';
 import OfferCard from '../../components/offer-card/offer-card';
-import {Offer} from '../../types/offers';
 
 type OffersListProps = {
   offers: Offer[],
   isFavoritePage: boolean,
-  hoverHandler?: (offer?: Offer) => void,
+  hoverHandler: (offer?:Offer) => void,
 };
 
-function OffersList({offers, isFavoritePage, hoverHandler}:OffersListProps): JSX.Element {
+function OffersList({offers, isFavoritePage, hoverHandler}: OffersListProps): JSX.Element {
   return (
     <>
       {offers.map((offer: Offer) => {
-        const keyValue = `${offer.id}-${offer.host.name}`;
+        const key = offer.host.name;
 
-        return (
-          hoverHandler !== undefined ?
-            <OfferCard key={keyValue} offer={offer} isFavoritePage={isFavoritePage} onStateChange={() => hoverHandler(offer)} onStateReset={() => hoverHandler()}/> :
-            <OfferCard key={keyValue} offer={offer} isFavoritePage={isFavoritePage}/>
+        return(
+          <Fragment key={key}>
+            <OfferCard
+              offer={offer}
+              isFavoritePage={isFavoritePage}
+              onStateChange={() => hoverHandler(offer)}
+              onStateReset={() => hoverHandler()}
+            />
+          </Fragment>
```

Comment:
```suggestion
            <OfferCard
              key={key}
              offer={offer}
              isFavoritePage={isFavoritePage}
              onStateChange={() => hoverHandler(offer)}
              onStateReset={() => hoverHandler()}
            />
```

Тебе не нужен тут фрагмент, у тебя же сразу возвращается компонент